ddl/ddl.go

```diff
@@ -431,19 +432,29 @@ func (d *ddl) buildColumnsAndConstraints(ctx context.Context, colDefs []*ast.Col
 	return cols, constraints, nil
 }
 
-func (d *ddl) buildColumnAndConstraint(ctx context.Context, offset int,
-	colDef *ast.ColumnDef) (*table.Column, []*ast.Constraint, error) {
-	// Set charset.
-	if len(colDef.Tp.Charset) == 0 {
-		switch colDef.Tp.Tp {
+func (d *ddl) setCharsetFlenDecimal(tp *types.FieldType) {
```

Comment:
It only sets Flen and Decimal, charset and collation is not set in this function.